llama_index/extractors/interface.py

```diff
@@ -113,6 +113,9 @@ async def aprocess_nodes(
         else:
             new_nodes = [deepcopy(node) for node in nodes]
 
+        if hasattr(self, "filter_nodes"):
+            new_nodes = self.filter_nodes(new_nodes)
```

Comment:
I think this means, if I ran `title_extractor(nodes)`, then it might return less nodes than I had in the input (which I think is unexpected?)

I think the expectation would be to run on the first 5 nodes (or whatever the user set) and copy the generated title from those first 5 nodes  to all nodes (or at least nodes with the same ref doc id),  all within the title extractor

Reply:
oh I see. Will fix.